server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Email.scala

```diff
@@ -60,8 +64,21 @@ object Email {
       refinedValue => refinedValue)
   }
 
-  def from(message: (MessageId, Seq[MessageResult])): Email =
-    Email(message._1, sanitizeSize(message._2.head.getSize))
+  def from(message: (MessageId, Seq[MessageResult])): Try[Email] = {
+    val firstMessage: MessageResult = message._2.head
+
+    val defaultMessageBuilder = new DefaultMessageBuilder
+    defaultMessageBuilder.setMimeEntityConfig(MimeConfig.PERMISSIVE)
+    defaultMessageBuilder.setDecodeMonitor(DecodeMonitor.SILENT)
+    val mime4JMessage: Message = defaultMessageBuilder.parseMessage(firstMessage.getFullContent.getInputStream)
```

Comment:
i am surprised that this type of method doesn't throw ? Should it return a Try[Message] ?